src/main/java/algorithms/minimumSpanningTree/kruskal/DisjointSet.java

```diff
@@ -0,0 +1,59 @@
+package algorithms.minimumSpanningTree.kruskal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of quick-find structure; Turns a list of objects into a data structure that supports union operations
+ */
+public class DisjointSet {
+    private final Map<Node, String> identifier;
+
+    /**
+     * Constructor to initialize Disjoint Set with a known list of listOfNodes.
+     * @param listOfNodes
+     */
+    public DisjointSet(Node[] listOfNodes) {
```

Comment:
nice, but we should align our implementation of disjoint set. There already is an existing one which u can either import or copy-paste (maybe this is preferable? to be explicit).

Ok but i did notice a concern as i was looking through. Both urs and my implementation does not handle duplicates well. I will tweak the official DisjointSet implementation of our repo to identify by memory address instead.